DEBUGGING.md

```diff
@@ -0,0 +1,71 @@
+# Debugging the OpenTofu code
+
+There are various ways to debug the OpenTofu code. There is no ultimate "right" answer, this document intends to collect some of those ways. The order of debugging techniques is completely random.
+
+If you would like to contribute to this debugging guide, please [create a GitHub issue](https://github.com/opentofu/opentofu/issues/new/choose) and propose the enhancement. After that, you can create a pull request and reference this issue in your PR.
+
+For further information on contributing to the code, please refer to the [CONTRIBUTING.md](./CONTRIBUTING.md) file.
+
+<!-- TOC -->
+
+- [Using the debug-opentofu script](#using-the-debug-opentofu-script)
+- [Using spew](#using-spew)
+- [Using VsCode](#using-vscode)
+
+<!-- /TOC -->
+
+## Using the debug-opentofu script
+
+[debug-opentofu](./scripts/debug-opentofu) is a helper script to launch OpenTofu inside the "dlv" debugger, configured to await a remote debugging connection on port 2345. For more details on how to use this script, please refer to the documentation at the beginning of this script.
```

Comment:
**(suggestion, non-blocking)** Hey! WDYT about linking [delve](https://github.com/go-delve/delve/tree/master)?

**(suggestion, nitpick)** Also, it makes sense to replace the link in the script's [comments](https://github.com/opentofu/opentofu/blob/03c8f6cebd2dcca4b8a1c6439322cc02395d46c1/scripts/debug-opentofu#L10):

_From_:
```
#    https://github.com/derekparker/delve/tree/master/Documentation/installation
```

_To_:
```
#    https://github.com/go-delve/delve/tree/master/Documentation/installation
```

WDYT?

Reply:
Sure, both are good suggestions, thanks :)

Reply:
Both suggestions are applied, resolving the conversation. Please don't hesitate to reopen if I missed anything.